documentation/src/main/asciidoc/topics/search.adoc

```diff
@@ -0,0 +1,1093 @@
+
+== Overview
+
+Searching is possible both in link:#query_library[library] and link:#query_remote[client/server mode] (for Java, C#, Node.js and other clients), and {brandname} can index data using link:http://lucene.apache.org/[Apache Lucene], offering an efficient link:https://en.wikipedia.org/wiki/Full-text_search[full-text] capable search engine in order to cover a wide range of data retrieval use cases.
+
+Indexing configuration relies on a schema definition, and for that {brandname} can use annotated Java classes when in library mode, and protobuf schemas for remote clients. By standardizing on protobuf, {brandname} allows full interoperability between Java and non-Java clients.
+
+{brandname} has its own query language called link:#query_ickle[Ickle], which is string-based and adds support for full-text searching. Ickle support searches over indexed data, partially indexed
+data or non-indexed data.
+
+Finally, {brandname} has support for link:#query_continuous[Continuous Queries], which works in a reverse manner to the other APIs: instead of creating, executing a query
+and obtain results, it allows a client to register queries that will be evaluated continuously as data in the cluster changes, generating notifications
+whenever the changed data matches the queries.
+
+== Indexing Entry Values
+
+Indexing entry values in {brandname} caches dramatically improves search performance and allows you to perform full-text queries. However, indexing can degrade write throughput for {brandname} clusters. For this reason you should plan to use strategies to optimise query performance, depending on the cache mode and your use case. More information on link:#query_performance[query performance guide].
+
+=== Configuration
+
+To enable indexing via XML, you need to add the `<indexing>` element to your cache configuration, specify the entities that are indexed and optionally pass additional properties.
+
+[NOTE]
+====
+The presence of an `<indexing>` element which omits the `enabled` attribute will auto-enable indexing for your
+convenience, even though the default value of the `enabled` attribute is defined as `"false"` in the XSD schema.
+====
+
+.Declaratively
+[source,xml,options="nowrap",subs=attributes+]
+----
+include::config_examples/indexing.xml[]
+----
+
+.Programmatically
+[source,java]
+----
+import org.infinispan.configuration.cache.*;
+
+ConfigurationBuilder cacheCfg = ...
+cacheCfg.indexing().enable()
+            .addIndexedEntity(Book.class)
+      .addProperty("property name", "propery value")
+----
+
+The presence of an `<indexing>` element which omits the `enabled` attribute will auto-enable indexing for your
+convenience, even though the default value of the `enabled` attribute is defined as `"false"` in the XSD schema.
+This behaviour exists only on schemas starting with version 11.
```

Comment:
Duplicate with the note above. I'd also move "In the programmatic config, `enabled()` must be used." into the note.